File: src/ProjectR/content/scripts/spells/FireWheel.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Fire Wheel"
end

function GetDescription()
  return "Spin around shrouded in fire and attack your enemies."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .84
end

function GetDelay()
  return .35
end

function GetSpellType()
  return Composite
end

function GetMasteries()
  return { FIR }
end

function SpellEffect()
  local damage = ((4 * aAD + 4 * aMD) * (aFIR/100) - (dDEF + dMR)) * (100/dFIR)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/FireWheel.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class FireWheel : SpellScriptBase
    {
        public override string Name { get { return "Fire Wheel"; } }
        public override string Description { get { return "Spin around shrouded in fire and attack your enemies."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.FIR, }; } }
        public override SpellType SpellType { get { return SpellType.Composite; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 84; } }
        public override double Delay { get { return .35; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = ((4 * cAD + 4 * cMD) * (cFIR / 100) - (tDEF + tMR)) * (100 / tFIR);
            DealDamage(damage);
        }
    }
}